### PR TITLE
sage: 9.8 -> 10.0

### PR DESCRIPTION
**Pull request by collares**
###### Description of changes

Includes updates to Singular, PARI, Giac and eclib. I didn't upgrade Giac to the latest version (1.9.0-49) because it was failing tests.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/math/sage/sagelib.nix
Comment (SuperSandro2000) [inline pkgs/applications/science/math/sage/sagelib.nix:113]: This has nothing to do with cross compilation, right? If so maybe propagatedNativeBuildInputs is a better fit?

Comment (collares) [reply pkgs/applications/science/math/sage/sagelib.nix:113]: That's a good point, I will check. Thanks!

Comment (symphorien) [reply pkgs/applications/science/math/sage/sagelib.nix:113]: non-python dependencies should probably not be propagated at all.

**Comment by collares** [comment]
@ofborg build sageWithDoc

File: pkgs/applications/science/math/sage/sagelib.nix
Comment (collares) [reply pkgs/applications/science/math/sage/sagelib.nix:113]: Technically, this isn't a regression from this PR, as the above changes just split previously-existing dependencies into two categories according to whether they're listed in sagelib's `requirements.txt` or not. Since most of those dependencies are binaries which will be executed at runtime by Sage, a proper split will require a bit of tweaking. I will do it in a follow-up PR.

**Comment by collares** [comment]
@ofborg build giac-with-xcas sageWithDoc

**Comment by collares** [comment]
@ofborg build giac-with-xcas sageWithDoc

**Comment by collares** [comment]
@ofborg build giac-with-xcas sageWithDoc

File: pkgs/applications/science/math/sage/sage-src.nix
Comment (collares) [inline pkgs/applications/science/math/sage/sage-src.nix:36]: I am considering just running `src/doc/bootstrap`, it would remove a few hacks (and it doesn't require running `src/bootstrap`, which is good).

Comment (7c6f434c) [reply pkgs/applications/science/math/sage/sage-src.nix:36]: Are there any drawbacks to using that `bootstrap` script?

Comment (collares) [reply pkgs/applications/science/math/sage/sage-src.nix]: The root bootstrap script requires network, so it is a no go. The doc bootstrap script, which generates a page of documentation for each external package required by Sage, expects to be called from the root one and reuses some of its infrastructure to list packages, but with a couple of hacks it runs fine.

The main drawback is conceptual: it uses spkg information from Sage-the-distribution (the `build` directory), but it's not too bad since it's just package names and versions which are contained in the main Sage repository. It's just a bit unfortunate that part of the upstream documentation (the section on how to use the GitHub CLI) now has links to the spkg section of the manual, requiring us to either not ignore the `build` directory or patch the references away.

I've pushed a commit that runs the doc bootstrap script (but not the root one). The previous version of the update just manually patched the (single) spkg reference away from the developer documentation. Either solution is fine by me.

File: pkgs/applications/science/math/sage/sagedoc.nix
Comment (collares) [inline pkgs/applications/science/math/sage/sagedoc.nix:42]: These hacky lines are the practical downside to running the bootstrap script. I kept them contained to `sagedoc.nix`.

**Comment by collares** [comment]
The aarch64 test timeout is rare, not specific to aarch64 and preexisting (sagemath/sage#33161).

**Review by omasanori — APPROVED** [review]
This contains an update to the latest upstream release, with massive reduction of downstream changes. I found no new issues here while skimming patches, but waiting for additional reviews by someone else for few days may be good.

By the way, I missed the current development status on 23.05. If ZHF is ongoing, we would land this PR on unstable after we finished 23.05 and backport it then to avoid mess?

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
Currently sage is failing for aarch64-linux and is not even present in darwin hydra. Going to merge this now since sage is always a difficult child and requires more attention to keep working.

**Pull request merged**
